### PR TITLE
Create show_supertypes function

### PR DESCRIPTION
**Pull request by omus**
A colleague of mine wanted to visualize the type hierarchy so I came up with this. I would like to potentially work it into `show` itself but I was unsure how I should integrate this.

### CODE REVIEW

**Comment by StefanKarpinski** [comment]
Maybe part of the `?T` output for a type?

**Comment by omus** [comment]
Added `show_supertypes` into type help:
```
help?> Integer
search: Integer isinteger

  No documentation found.

  Summary:

  abstract type Integer <: Real

  Subtypes:

  BigInt
  Bool
  Signed
  Unsigned

  Supertype Hierarchy:

  Integer <: Real <: Number <: Any
```

**Comment by omus** [comment]
Additionally, limited the number of displayed subtypes in the type help:
```
help?> Any
search: Any any ANY any! listenany randcycle code_warntype ReadOnlyMemoryError @code_warntype isreadonly

  No documentation found.

  Summary:

  abstract type Any <: Any

  Subtypes:

  AbstractArray
  AbstractChannel
  AbstractRNG
  AbstractSerializer
  AbstractSet
  AbstractString
  Any
  Associative
  Base.AbstractCartesianIndex
  Base.AbstractCmd
  Base.AsyncCollector
  Base.AsyncCollectorState
  Base.AsyncCondition
  Base.AsyncGenerator
  Base.AsyncGeneratorState
  Base.BaseDocs.Keyword
  Base.Cartesian.LReplace
  Base.ChannelIterState
  Base.CodegenHooks
  Base.CodegenParams
  ...
```

File: base/docs/Docs.jl [hunk 1]
```diff
@@ -395,9 +395,18 @@ function summarize(io::IO, T::DataType, binding)
     if !isempty(subtypes(T))
         println(io, "**Subtypes:**")
         println(io, "```")
-        for t in subtypes(T)
+        sub_types = subtypes(T)
+        for t in Iterators.take(sub_types, 20)
```
Comment (tkelman) [inline base/docs/Docs.jl]: do we hardcode limits like this elsewhere? could maybe be an iocontext property eventually

**Comment by omus** [comment]
I removed the limiting the amount of show subtypes in the help. I'll make another PR for that.

**Review by ararslan — APPROVED** [review]
Nice, I like it. 👍

File: base/docs/Docs.jl [hunk 2]
```diff
@@ -400,6 +400,13 @@ function summarize(io::IO, T::DataType, binding)
         end
         println(io, "```")
     end
+    if supertype(T) != Any
+        println(io, "**Supertype Hierarchy:**")
```
Comment (musm) [inline base/docs/Docs.jl]: should these both be capitalized?

Comment (omus) [inline base/docs/Docs.jl]: I prefer using title case for headings

**Comment by omus** [comment]
I'll merge this later today unless there are objections.

**Comment by TotalVerb** [comment]
It's a little sad that this is disabled if the type is documented. Is there a way we can get this into help regardless of whether the type is documented?

**Comment by StefanKarpinski** [comment]
Personally, I'm in favor of more complete automatic help output for things, including supertypes for types and methods for functions.

**Comment by yuyichao** [comment]
> more complete automatic help output for things, including supertypes for types and methods for functions.

That's exactly what's causing https://github.com/JuliaLang/julia/issues/20904 though....

**Comment by StefanKarpinski** [comment]
Yes, well, there are more direct ways to fix that bug than just avoiding triggering it.

**Comment by yuyichao** [comment]
Ture. What I'm trying to say is that the automatically generated doc isn't very helpful sometimes and it should be possible to manually write documents that are much better. `methods`, `subtypes`, `methodswith` are certainly useful sometimes but I also find it not helping most of the time especially if they take a large fraction of screen space making it harder to find more useful information. I assume this is roughly the rationale behind the current behavior. Integrating them better with the doc system would be good. (paging, filtering, highting, mentioning their existence in the doc of doc system, whatever....) but I don't think we should unconditionally include them.

OTOH though for this particular case, the super type list is likely short so including **this** unconditionally might be acceptable.

**Pull request merged**